chapters/packages.tex

```diff
@@ -326,3 +326,117 @@ \section{External resources}\label{external-resources}
+
+The header is marked with an empty \emph{msgid} entry and looks like this:
+\begin{lstlisting}
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-03-15 10:52+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+In the body, directly following the header, for each descriptive text there is an entry of the form
+\begin{lstlisting}
+#: <file name>:<line number>
+msgctxt "<context identifier>"
+msgid "<descriptive text>"
+msgstr "<translation>"
+\end{lstlisting}
+Only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used.
+
+At first there can be an optional comment describing the location (file name and line number) of the text to translate.
+ Multiple occurences of the same string can be listed here, separated by space.
+
+Then, the \lstinline!<context identifier>! behind the keyword \lstinline!msgctxt! is the full name of the Modelica class (e.g., \lstinline!Modelica.Blocks.Math.Sin!) where the text appears in.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+After the \lstinline!msgid! keyword the text string which shall be translated follows.
+It should contain the original string from the Modelica text representation.
+Since in Modelica control sequences also start with a backslash and another backslash is used to use sequences literally or to hide double quotes, no change is required here.
+But Modelica allows strings to go over more than one line, gettext does not.
+Therefore, line breaks in Modelica must be encoded with \lstinline!"\n"! for gettext.
+
+\begin{nonnormative}
+In order to avoid long lines, strings may be split into smaller parts given on consecutive lines, e.g.
+\begin{lstlisting}
+"A
+B"
+\end{lstlisting}
+is converted to
+\begin{lstlisting}
+"A\n"
+"B"
+\end{lstlisting}
+These strings are concatenated when read.
+Please regard that if a \lstinline!msgid! string is given more than once in the same context, all occurrences are translated with the same (last) translation!
+\end{nonnormative}
+
+The keyword \lstinline!msgstr! is followed by the translation of \lstinline!msgid!.
+With regard to control characters and line breaks, the same rules apply as for \lstinline!msgid!.
+In the template file this string is empty by definition.
+If this is empty in a language specific file the contents of \lstinline!msgid! may be used instead.
```

Comment:
More repetition of gettext specification to be marked non-normative or removed.

Reply:
This is now nonnormative.

Reply:
This is now nonnormative.

Reply:
Yes.  Resolving.